addons/website/static/src/snippets/s_social_media/options.js

```diff
@@ -0,0 +1,214 @@
+/** @odoo-module **/
+
+import options from 'web_editor.snippets.options';
+
+options.registry.SocialMedia = options.Class.extend({
+    /**
+     * @override
+     */
+    async willStart() {
+        this._super(...arguments);
+        let websiteId;
+        this.trigger_up('context_get', {
+            callback: function (ctx) {
+                websiteId = ctx['website_id'];
+            },
+        });
+        // Fetch URLs for db links.
+        const [res] = await this._rpc({
+            model: 'website',
+            method: 'read',
+            args: [websiteId, ['social_facebook', 'social_twitter', 'social_youtube',
+                'social_instagram', 'social_linkedin', 'social_github']],
+        });
+        delete res.id;
+        for (const socialMedia of Object.entries(res)) {
+            const linkEl = this.$target[0].querySelector(`a[data-social-id="${socialMedia[0]}"]`);
+            if (linkEl) {
+                linkEl.dataset.dbUrl = socialMedia[1] ? socialMedia[1] : '';
+            }
+        }
+    },
+    /**
+     * @override
+     */
+    cleanForSave() {
+        const changed = {};
+        const anchorsEls = this.$target[0].querySelectorAll('a[data-social-id]');
+        for (const anchorEl of anchorsEls) {
+            if (anchorEl.hasAttribute('data-db-url')) {
+                changed[anchorEl.dataset.socialId] = anchorEl.dataset.dbUrl;
+                delete anchorEl.dataset.dbUrl;
+            }
+        }
+        // Update the DB links.
+        let websiteId;
+        this.trigger_up('context_get', {
+            callback: function (ctx) {
+                websiteId = ctx['website_id'];
+            },
+        });
+        this._rpc({
```

Comment:
Should cleanForSave be async here and the rpc be awaited?

Reply:
I am not 100% sure myself, hence my question :-)

The `cleanForSave()` is being awaited in the `destroyEditors` (or smth like that), so I was surprised to see unwaited RPCs. From what I understand, it means that everything could continue without waiting this RPC, ultimately destroying/saving before the RPC is actually done.

I would imagine the page could then reload (eg you changed a media URL and then an option that performed a reload), then the reload could actually be done before the write website RPC is done (most likely it should not), and you have wrong values since the read rpc on start will fetch wrong values.

Not a really big deal, but just illustrate the error in the code as far as I understand it.

You can ignore and leave this review open for now, I'll triple check it.

Reply:
Note that because there is a similar occurrence elsewhere in the code it doesn't mean it is not wrongly coded and/or introduce errors.
I was just wondering conceptually here if there is not a promise misbehavior (maybe not).